src/index.js

```diff
@@ -174,9 +174,12 @@ class Dropzone extends React.Component {
     })
 
     if (!multiple) {
-      // if not in multi mode add any extra accepted files to rejected.
-      // This will allow end users to easily ignore a multi file drop in "single" mode.
-      rejectedFiles.push(...acceptedFiles.splice(1))
+      // Only allow one file to be accepted if multiple=false
+      if (this.state.draggedFiles.length === 1) {
+        rejectedFiles.push(...acceptedFiles.splice(1))
+      } else {
+        rejectedFiles.push(...acceptedFiles.splice(0))
```

Comment:
Why is this `splice` needed here if you already doing `...`?

Reply:
i thought the spread operator in javascript just turns items in an array to objects with a `key: value` pair? and as i found out splice used with 1 argument is a little hacky? (not sure though)

Reply:
> i thought the spread operator in javascript just turns items in an array to objects with a key: value pair? 

No, this is not what spread is doing: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator

There is no need in `.splice(0)` since it's literally what `...` is doing. I'm not even sure we need `...` in front of the first one — seems like a redundant code to me. Please check this.